src/hssm/prior.py

```diff
@@ -0,0 +1,135 @@
+"""bmb.Prior on steroids.
+
+This class is a subclass of bmb.Prior, which is used to temporarily represent
+a prior distribution until model build time. This class retains all functionalities
+bmb.Prior but adds the following:
+
+1. The ability to represent a truncated prior.
+2. The ability to still print out the prior before the truncation.
+3. The ability to shorten the output of bmb.Prior.
+"""
+
+from __future__ import annotations
+
+from typing import Callable
+
+import bambi as bmb
+import numpy as np
+import pymc as pm
+from bambi.backend.utils import get_distribution
+from bambi.priors.prior import format_arg
+
+pymc_dist_args = ["rng", "initval", "dims", "observed", "total_size", "transform"]
+
+
+# mypy: disable-error-code="has-type"
+class Prior(bmb.Prior):
+    """Create a Prior object that extend the functionalities of bmb.Prior.
+
+    Parameters
+    ----------
+    bmb
+        _description_
+    """
+
+    def __init__(
+        self,
+        name: str,
+        auto_scale: bool = True,
+        dist: Callable | None = None,
+        bounds: tuple[float, float] | None = None,
+        **kwargs,
+    ):
+        bmb.Prior.__init__(self, name, auto_scale, dist, **kwargs)
+        self.is_truncated = False
+        self.bounds = bounds
+
+        if self.bounds is not None:
+            assert self.dist is None, (
+                "We cannot bound a prior defined with the `dist` argument. The "
+                + "`dist` and `bounds` arguments cannot both be supplied."
+            )
+            lower, upper = self.bounds
+            if np.isinf(lower) and np.isinf(upper):
+                return
+
+            self.is_truncated = True
+            self.dist = _make_truncated_dist(self.name, lower, upper, **self.args)
+            self._args = self.args.copy()
+            self.args: dict = {}
```

Comment:
overwriting the original .args ?

Reply:
what kind of issues are we talking about?
Just asking because .args is something that is expected to be carried by a Bambi prior right?

Reply:
To make sure there aren't any unintended side-effects introduced by skipping this attribute (making it empty)

Reply:
I see ok